Runtime/AvatarCreator/UI/Elements/AssetSelectionElement.cs

```diff
@@ -0,0 +1,62 @@
+using System.Linq;
+using System.Threading.Tasks;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+
+    /// <summary>
+    /// Manages the selection of assets for a given body type, gender, and asset type.
+    /// This class is responsible for loading asset data and creating corresponding buttons for each asset.
+    /// It extends the functionality of SelectionElement to handle PartnerAsset-specific interactions.
+    /// </summary>
+    public class AssetSelectionElement : SelectionElement
+    {
+        [Header("Properties")]
+        [SerializeField] private BodyType bodyType = BodyType.FullBody;
+        [SerializeField] private AssetType assetType;
+        [SerializeField] private int iconSize = 64;
+
+        private PartnerAsset[] partnerAssets;
+        private AssetAPIRequests partnerAssetsRequests;
+
+        private void Awake()
+        {
+            partnerAssetsRequests = new AssetAPIRequests(CoreSettingsHandler.CoreSettings.AppId);
+        }
+
+        public void SetBodyType(BodyType bodyType)
+        {
+            this.bodyType = bodyType;
+        }
+
+        /// <summary>
+        /// Asynchronously loads partner asset data based on the specified asset type, body type, and gender.
+        /// This method updates the internal partnerAssets collection with the fetched data.
+        /// </summary>
+        /// <param name="bodyType">The body type to filter the assets and determine the type of avatar to load.</param>
+        /// <param name="gender">The gender to filter the assets and determine which skeleton will be loaded.</param>
+        /// <returns>A Task representing the asynchronous operation of fetching and loading the partner asset data.</returns>
+        public async Task LoadTemplateData(OutfitGender gender)
```

Comment:
This method's name should be changed, as it isn't template data in this context.

Reply:
yep! good catch

Reply:
updated